src/compiler/scala/tools/nsc/typechecker/ConstantFolder.scala

```diff
@@ -34,16 +34,11 @@ abstract class ConstantFolder {
     case _ => null
   })
 
-  private def fold(tree: Tree, compX: => Constant): Tree =
-    try {
-      val x = compX
-      if ((x ne null) && x.tag != UnitTag) tree setType ConstantType(x)
-      else tree
-    } catch {
-      case _: ArithmeticException => tree   // the code will crash at runtime,
-                                           // but that is better than the
-                                           // compiler itself crashing
-    }
```

Comment:
I'm not sure why the warning isn't here instead. Sorry I'm distracted by day job...

Reply:
In my distraction, I didn't noticed that the check file wasn't checked under pos.